File: app/src/main/java/com/imangazalievm/bubbble/domain/interactors/ShotDetailsInteractor.java
package com.imangazalievm.bubbble.domain.interactors;


import com.imangazalievm.bubbble.domain.models.Comment;
import com.imangazalievm.bubbble.domain.models.Shot;
import com.imangazalievm.bubbble.domain.models.ShotCommentsRequestParams;
import com.imangazalievm.bubbble.domain.repository.CommentsRepository;
import com.imangazalievm.bubbble.domain.repository.ImagesRepository;
import com.imangazalievm.bubbble.domain.repository.ShotsRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Completable;
import io.reactivex.Single;

public class ShotDetailsInteractor {

    private ShotsRepository shotsRepository;
    private CommentsRepository commentsRepository;
    private ImagesRepository imagesRepository;

    @Inject
    public ShotDetailsInteractor(ShotsRepository shotsRepository,
                                 CommentsRepository commentsRepository,
                                 ImagesRepository imagesRepository) {
        this.shotsRepository = shotsRepository;
        this.commentsRepository = commentsRepository;
        this.imagesRepository = imagesRepository;
    }

    public Single<Shot> getShot(long shotId) {
        return shotsRepository.getShot(shotId);
    }

    public Single<List<Comment>> getShotComments(ShotCommentsRequestParams shotCommentsRequestParams) {
        return commentsRepository.getComments(shotCommentsRequestParams);
    }

    public Completable saveImage(String imageUrl) {
        return imagesRepository.saveImage(imageUrl);
    }

}

File: app/src/main/java/com/bubbble/domain/shotdetails/ShotDetailsInteractor.kt
package com.bubbble.domain.shotdetails

import com.bubbble.core.models.Comment
import com.bubbble.core.models.shot.Shot
import com.bubbble.core.models.shot.ShotCommentsParams
import com.bubbble.data.shots.CommentsRepository
import com.bubbble.data.shots.ImagesRepository
import com.bubbble.data.shots.ShotsRepository
import javax.inject.Inject

class ShotDetailsInteractor @Inject constructor(
    private val shotsRepository: ShotsRepository,
    private val commentsRepository: CommentsRepository,
    private val imagesRepository: ImagesRepository
) {

    suspend fun getShot(shotId: Long): Shot {
        return shotsRepository.getShot(shotId)
    }

    suspend fun getShotComments(params: ShotCommentsParams): List<Comment> {
        return commentsRepository.getComments(params)
    }

    suspend fun saveImage(imageUrl: String) {
        return imagesRepository.saveImage(imageUrl)
    }

}